libft/tests/strings/ft_split_skip_delim_test.c

```diff
@@ -0,0 +1,125 @@
+/* ************************************************************************** */
+/*                                                                            */
+/*                                                        :::      ::::::::   */
+/*   ft_split_skip_delim_test.c                         :+:      :+:    :+:   */
+/*                                                    +:+ +:+         +:+     */
+/*   By: taretiuk <taretiuk@student.42.fr>          +#+  +:+       +#+        */
+/*                                                +#+#+#+#+#+   +#+           */
+/*   Created: 2024/10/23 16:24:51 by taretiuk          #+#    #+#             */
+/*   Updated: 2024/11/04 05:53:30 by taretiuk         ###   ########.fr       */
+/*                                                                            */
+/* ************************************************************************** */
+
+#include "../../libft.h"
+#include "tests.h"
+
+#define NUM_TEST_CASES 12
+#define MAX_ARGS 16
+
+typedef struct s_string
+{
+	char	*str;
+}	t_string;
+
+typedef struct s_string_array
+{
+	t_string	*strs;
+	int			error;
+	size_t		count;
+}	t_strings;
+
+static t_strings	create_string_array()
+{
+	t_strings	str_array;
+	str_array.count = 12;
+	str_array.error = 0;
+	str_array.strs = ft_calloc_if(sizeof(t_string) * str_array.count, 1);
+	assert(str_array.strs != NULL);
+	for (int i = 0; i < (int)str_array.count; i++)
+	{
+		str_array.strs[i].str = NULL;
+	}
+	str_array.strs[0].str = strdup("cat ( file1.txt file2.txt ) | grep \"keyword\"");
+	str_array.strs[1].str = strdup("cat < input.txt && echo | grep \"Done\"");
+	str_array.strs[2].str = strdup("grep \"error\" log.txt || echo \"No errors    found\"");
+	str_array.strs[3].str = strdup("gcc main.c -o program >> build.log");
+	str_array.strs[4].str = strdup("find . -name \"*.txt\" | xargs rm -rf || fer && def & echo \"Hello\"");
+	str_array.strs[5].str = strdup("echo \"Compiling...\" && gcc main.c -o program && ./program");
+	str_array.strs[6].str = strdup("mkdir new_dir && cd new_dir && touch file.txt");
+	str_array.strs[7].str = strdup("mv new_dir old_dir");
+	str_array.strs[8].str = strdup("&&");
+	str_array.strs[9].str = strdup("echo \"ls -l && cd ..\"");
+	str_array.strs[10].str = strdup(")(())(");
+	str_array.strs[11].str = strdup("");
+	return (str_array);
+}
+
+static void	free_tokens(char **tokens)
+{
+	if (!tokens)
+		return;
+	for (int i = 0; tokens[i] != NULL; i++)
+	{
+		free(tokens[i]);
+		tokens[i] = NULL;
+	}
+	free(tokens);
+}
+
+static void	free_string_array(t_string *strs, size_t count) 
+{
+	if (!strs)
+		return;
+
+	for (size_t i = 0; i < count; i++) 
+	{
+		free(strs[i].str);
+	}
+	free(strs);
+}
+
+void ft_split_skip_delim_test()
+{
+	t_strings	strings_arr = create_string_array();
+	char		**tokens;
+	int			sz = 0;
+
+	char *t[NUM_TEST_CASES][MAX_ARGS] =
+	{
+		{"cat", "(", "file1.txt", "file2.txt", ")", "|", "grep", "\"keyword\"", NULL},
```

Comment:
It's somewhat hard to match test inputs to expected results if they are stored in different structures and defined in different places (not in one/two consecutive lines). Please consider using something like [t_testcase](https://github.com/ioannnikulin/minishell/pull/91/files#diff-96f3524bb234b2822aec0dd24e1d17a99ef47a51202bfa6ae88820b3f17ad7b8R15) next time.
Not a blocker, mark resolved when read.

Reply:
thank you, it's hard to understand right now your tests, maybe i'll ask you next time to explain